conans/test/integration/diamond_test.py

```diff
@@ -20,8 +20,26 @@ def setUp(self):
         test_server = TestServer(
                                  [],  # write permissions
                                  users={"lasote": "mypass"})  # exported users and passwords
-        self.servers = {"default": test_server}
-        self.conan = TestClient(servers=self.servers, users={"default": [("lasote", "mypass")]})
+        servers = {"default": test_server}
+        conan = TestClient(servers=servers, users={"default": [("lasote", "mypass")]})
+        self.diamond_tester = DiamondTester(self, conan, servers)
+
+    def diamond_cmake_test(self):
+        self.diamond_tester.test(use_cmake=True)
+
+    def diamond_cmake_targets_test(self):
+        self.diamond_tester.test(use_cmake=True, cmake_targets=True)
+
+    def diamond_default_test(self):
+        self.diamond_tester.test(use_cmake=False)
+
+
+class DiamondTester(object):
```

Comment:
I don't see the advantage of the DiamondTester

Reply:
It's used in the `conan-io/test` to build a diamond with mingw